core/src/main/java/org/ironjacamar/core/spi/classloading/ClassLoaderPlugin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  IronJacamar, a Java EE Connector Architecture implementation
+ *  Copyright 2016, Red Hat Inc, and individual contributors
+ *  as indicated by the @author tags. See the copyright.txt file in the
+ *  distribution for a full listing of individual contributors.
+ *
+ *  This is free software; you can redistribute it and/or modify it
+ *  under the terms of the Eclipse Public License 1.0 as
+ *  published by the Free Software Foundation.
+ *
+ *  This software is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Eclipse
+ *  Public License for more details.
+ *
+ *  You should have received a copy of the Eclipse Public License
+ *  along with this software; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ *  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.ironjacamar.core.spi.classloading;
+
+/** The SPI for class loading
+ *
+ * @author <a href="mailto:stefano.maestri@ironjacamar.org">Stefano Maestri</a>
+ */
+public interface ClassLoaderPlugin
+{
+   /**
+    * get the ClassLoader
+    * @return the class loader
+    */
+   ClassLoader getClassLoader();
```

Comment:
This should resolve a class instance, based upon the classloader used internally in the plugin, ala
```
public Class<?> loadClass(String className, String module, String slot) throws ...
```